dcicutils/ff_utils.py

```diff
@@ -457,65 +457,150 @@ def faceted_search(key=None, ff_env=None, item_type=None, **kwargs):
     return search_metadata(search, ff_env=ff_env, key=key)
 
 
-def get_associated_qc_metrics(uuid, key=None, ff_env=None,
-                              exclude_raw_files=True,
-                              exclude_supplementary_files=True):
+def fetch_files_qc_metrics(data, associated_files=['processed_files'], ignored_fields=True, key=None, ff_env=None):
     """
-    Given a uuid of an experiment set, return a dictionary of uuid : item
-    mappings of quality metric items.
+    Utility function to grap all the qc metrics from associated types of file such as:
+    'proccessed_files', 'other_processed_files', 'files'
+    inputs:
+        data: the metadata of a ExperimentSet or Experiment
+        associated_files: a list of the types of the files fields the qc metrics will be extracted from:
+            examples are = ['files', 'processed_files', 'other_processed_files']
+    Args:
+        ignored_fields: flag to ignore 4DN custom fields from the qc metric object
```

Comment:
If it's just a boolean, I might rename this argument to `ignore_typical_fields` or some other kind of thing that suggest you're not supposed to pass the fields but assume they'll be figured out for you. However, you could perhaps allow both and then could leave this variable name alone. See my comment below.